ssl/quic/quic_impl.c

```diff
@@ -4436,6 +4436,8 @@ SSL *ossl_quic_accept_connection(SSL *ssl, uint64_t flags)
 
 out:
     qctx_unlock(&ctx);
+    if (qc != NULL)
```

Comment:
I have two questions here:
   - why we need it here? 
   - is it OK to ignore error which may come from `ossl_quic_do_handshake()` here?

the way I understand the current code is that the `quic_do_handshake()` will get called with the next operation on SSL object returned by accept. Is it just kind of short-cut to initiate handshake early here? I would prefer to remove this change from this PR unless it is really needed here.

Reply:
in response to questions, the NULL check on line 4439 is because of the conditional on line 4401.  If ql_listen fails, we jump to the out label, and qc is guaranteed to be NUL, which will lead to a segfaul when dereferencing &qc->obj.ssl in the call to ossl_quic_do_handshake.

In regards to ignoring the error, I'm not sure, but nor am I sure what we should do in the event that do handshake fails here.

Reply:
sorry my question was abuut the whole change here, not just the single line.

the way I understand the current code the accept is supposed to return QUIC object. the caller then passes the object to `SSL_read()`/`SSL_write()/...` to initiate handshake eventually. those functions (write/read) are in much better position to indicate failure in handshake. my concern here is that we initiate handshake and ignore the error here. so si error state kept in object so the next call to `SSL_read/SSL_write` finds the error and can report it to caller properly?

I'm sorry my first comment was bit messy.

Reply:
ack, looking at it more closely, I think its actaully leftover from some early experiment I was doing, and its not needed for anything.  I'm confirming that removing it has no impact on the test now